devito/types/tensor.py

```diff
@@ -0,0 +1,598 @@
+from collections import OrderedDict
+
+import sympy
+import numpy as np
+from sympy.core.sympify import converter as sympify_converter
+from sympy.core.decorators import call_highest_priority
```

Comment:
I can smell the joy you had in this digging this up

Reply:
Haha yeah was fun